### PR TITLE
Config Testing, bugfixes, Mock data generation, e2e client tests + template adjustments

### PR DESCRIPTION
**Pull request by alexthemayers**

### CODE REVIEW

File: openapi/schema/schema.go [hunk 1]
```diff
@@ -6,8 +6,10 @@ import (
 	"path/filepath"
 )
 
-const PropertyKey = "Model"
-const ItemsKey = "9dce7025-4caf-415b-a4fd-e74e54d098cb"
+const (
+	PropertyKey = "5be9ff96-96ae-47b9-b878-1340b357f202"
```
Comment (SasSwart) [inline openapi/schema/schema.go:12]: Please document the reasoning behind why we made these UUIDs.

File: openapi/schema/schema.go [hunk 2]
```diff
@@ -82,3 +84,21 @@ func (s *Schema) IsRequired(property string) bool {
 	}
 	return false
 }
+func (s *Schema) GetName() []string {
```
Comment (SasSwart) [inline openapi/schema/schema.go:89]: Nice. And I mean it this time

File: render/go/renderer.go [hunk 3]
```diff
@@ -129,3 +126,52 @@ func isHttpStatusCode(s string) bool {
...
+	config.ConfigFilePath = "../render/go/config_goclient_test.yml"
+	config.Debug = true
+	return config.Data{
+		Template: config.Template{
+			Name: "go-client",
+		},
+		OpenAPIFile: "../openapi/test/fixtures/validation_no_refs.yaml",
+		OutputPath:  ".",
+	}
+}
```
Comment (SasSwart) [inline render/go/renderer.go:192]: Do these have to be here, or can they move to the appropriate `_test.go` file?

File: render/go/renderer_test.go [hunk 4]
```diff
@@ -179,3 +176,22 @@ func TestGolang_CleanFunctionString(t *testing.T) {
...
 	}
 }
+
+func TestGolang_ToTitle(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		expected string
+	}{
+		{},
```
Comment (SasSwart) [inline render/go/renderer_test.go]: No test cases yet?

File: test/mocks.go [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+	"github.com/sasswart/gin-in-a-can/openapi/request"
+	"github.com/sasswart/gin-in-a-can/openapi/response"
+	"github.com/sasswart/gin-in-a-can/openapi/schema"
+	"github.com/sasswart/gin-in-a-can/tree"
+	"net/http"
+)
+
+var md = tree.Metadata{"package": "testPackage", "some": "metadata"}
+
+func OpenAPITree() *openapi.OpenAPI {
```
Comment (SasSwart) [inline test/mocks.go:17]: Remind me why you aren't reading this from a fixture file?

File: render/go/renderer.go (hunk 3 above)
Comment (alexthemayers) [reply render/go/renderer.go:192]: we can't import from a `*_test.go` file and these are needed in multiple places. I originally had them in the test file and would rather not create a `helper` file.

File: test/mocks.go (hunk 5 above)
Comment (alexthemayers) [reply test/mocks.go:17]: These are mocks that are designed to test the renderer without involving the functionality that parses an openAPI spec file

File: openapi/schema/schema.go (hunk 2 above)
Comment (alexthemayers) [reply openapi/schema/schema.go:89]: The triplicate `if something == nil` checks bother me. I'd have used something like a preprocessor macro here if this were C. :/

But thanks!

Comment (SasSwart) [reply openapi/schema/schema.go:89]: quack quack

File: render/go/renderer_test.go (hunk 4 above)
Comment (alexthemayers) [reply render/go/renderer_test.go]: Sorted now

**Review by SasSwart — DISMISSED** [review]

**Review by SasSwart — APPROVED** [review]

**Pull request merged**